src/main/java/com/linkedin/kmf/apps/SingleClusterMonitor.java

```diff
@@ -61,17 +61,23 @@ public SingleClusterMonitor(Map<String, Object> props, String name) throws Excep
   @Override
   public void start() {
     _topicManagementService.start();
-    _produceService.start();
-    _consumeService.start();
-    LOG.info(_name + "/SingleClusterMonitor started");
+    CompletableFuture<Void> completableFuture = _topicManagementService.topicManagementReady();
+    completableFuture.thenRun(() -> {
+      try {
+        _produceService.start();
+        _consumeService.start();
+      } finally {
+        LOG.info(_name + "/SingleClusterMonitor started.");
+      }
```

Comment:
This shouldn't be in a finally block because it's possible produce service or the consumer service throws an exception and have not actually started.